scripts/lib/CIME/XML/component.py

```diff
@@ -116,14 +118,74 @@ def _get_value_match(self, node, attributes=None, exact_match=False):
         return match_value
 
     #pylint: disable=arguments-differ
-    def get_description(self, compsetname):
+    def get_description(self, compsetname, comp_class=None):
+        comp_class = comp_class.lower()
         rootnode = self.get_node("description")
-        desc_nodes = self.get_nodes("desc", root=rootnode)
         desc = ""
-        for node in desc_nodes:
-            compsetmatch = node.get("compset")
-            if re.search(compsetmatch, compsetname):
-                desc += node.text
+        desc_nodes = self.get_nodes("desc", root=rootnode)
+        if self.get_version() == 3.0:
+            expect(comp_class is not None,"comp_class argument required for version3 files")
+            modifier_mode = rootnode.get('modifier_mode')
+            if modifier_mode is None:
+                modifier_mode = '*'
+                expect(modifier_mode in ('*','1','?','+'),
+                       "Invalid modifier_mode {} in file {}".format(modifier_mode, self.filename))
+
+            optiondesc = {}
+            if comp_class == "forcing":
+                for node in desc_nodes:
+                    forcing = node.get('forcing')
+                    if forcing is not None and compsetname.startswith(forcing):
+                        desc = node.text
+                        return desc
+
```

Comment:
I'm thinking this should error out if you haven't found an appropriate forcing? Otherwise it's going to fall through to the code below, which I'm guessing is the wrong thing to do.